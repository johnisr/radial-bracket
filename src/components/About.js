import React from 'react';

const About = () => (
  <div>
    <h2>About</h2>
    <h4>This is the about page</h4>
  </div>
);

export default About;
