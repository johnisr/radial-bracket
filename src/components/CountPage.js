import React from 'react';
import { connect } from 'react-redux';

import { incrementCount, decrementCount, resetCount } from '../actions/count';

export class CountPage extends React.Component {
  handleIncrement = () => {
    this.props.increment(1);
  }
  handleDecrement = () => {
    this.props.decrement(1);
  }
  handleReset = () => {
    this.props.reset();
  }
  render() {
    return (
      <div className={'count'}>
        <h1 className={'heading-1 margin-bottom-large'}>Count: {this.props.count.count}</h1>
        <button className={'count__btn'} onClick={this.handleIncrement}>+</button>
        <button className={'count__btn'} onClick={this.handleDecrement}>-</button>
        <button className={'count__btn'} onClick={this.handleReset}>Reset</button>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    count: state.count,
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  increment: (number) => dispatch(incrementCount({incrementBy: number})),
  decrement: (number) => dispatch(decrementCount({decrementBy: number})),
  reset: () => dispatch(resetCount()),
});

export default connect(mapStateToProps, mapDispatchToProps)(CountPage);

