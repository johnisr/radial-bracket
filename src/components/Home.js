import React from 'react';

const Home = () => (
  <div>
    <h2>Home</h2>
    <h4>This is the homepage</h4>
  </div>
);

export default Home;

