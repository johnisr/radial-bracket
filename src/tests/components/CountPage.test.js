import React from 'react';
import { shallow } from 'enzyme';
import { CountPage } from '../../components/CountPage';

let increment, decrement, reset, wrapper;
beforeEach(() => {
  increment = jest.fn();
  decrement = jest.fn();
  reset = jest.fn();
  wrapper = shallow(
    <CountPage 
      increment={increment}
      decrement={decrement}
      reset={reset}
      count={{ count: 2 }}
    />)


});

it('should render CountPage', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle increment', () => {
  wrapper.find('.count__btn').at(0).simulate('click');
  expect(increment).toHaveBeenLastCalledWith(1);
});
it('should handle decrement', () => {
  wrapper.find('.count__btn').at(1).simulate('click');
  expect(decrement).toHaveBeenLastCalledWith(1);
});
it('should handle increment', () => {
  wrapper.find('.count__btn').at(2).simulate('click');
  expect(reset).toHaveBeenCalled();
});