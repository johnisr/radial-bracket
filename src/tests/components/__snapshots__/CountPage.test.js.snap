// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CountPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CountPage
    count={
        Object {
            "count": 2,
          }
    }
    decrement={[Function]}
    increment={[Function]}
    reset={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="heading-1 margin-bottom-large"
>
          Count: 
          2
</h1>,
        <button
          className="count__btn"
          onClick={[Function]}
>
          +
</button>,
        <button
          className="count__btn"
          onClick={[Function]}
>
          -
</button>,
        <button
          className="count__btn"
          onClick={[Function]}
>
          Reset
</button>,
      ],
      "className": "count",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Count: ",
            2,
          ],
          "className": "heading-1 margin-bottom-large",
        },
        "ref": null,
        "rendered": Array [
          "Count: ",
          2,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "count__btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "className": "count__btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset",
          "className": "count__btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reset",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="heading-1 margin-bottom-large"
>
            Count: 
            2
</h1>,
          <button
            className="count__btn"
            onClick={[Function]}
>
            +
</button>,
          <button
            className="count__btn"
            onClick={[Function]}
>
            -
</button>,
          <button
            className="count__btn"
            onClick={[Function]}
>
            Reset
</button>,
        ],
        "className": "count",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Count: ",
              2,
            ],
            "className": "heading-1 margin-bottom-large",
          },
          "ref": null,
          "rendered": Array [
            "Count: ",
            2,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "count__btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "className": "count__btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "className": "count__btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
